ports/stm32/mpconfigboard_common.h

```diff
@@ -221,3 +221,16 @@
 #endif
 
 #define MICROPY_HW_USES_BOOTLOADER (MICROPY_HW_VTOR != 0x08000000)
+
+// Heap start / end definitions
+#if MICROPY_HW_SDRAM_USE_FOR_HEAP
```

Comment:
Instead of adding this (yet-another) config variable, how about instead the mpconfigboard.h file just sets `MICROPY_HEAP_START` and `MICROPY_HEAP_END` itself when it has SDRAM?  So in those board configs with SDRAM they can add:
```c
#define MICROPY_HEAP_START sdram_start()
#define MICROPY_HEAP_END sdram_end()
```
Then it's easy for users to see how to update that to use less of the SDRAM, for example.